pkg/api/staking_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	stakingContractMock "github.com/ethersphere/bee/pkg/staking/stakingcontract/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestDepositStake(t *testing.T) {
+	t.Parallel()
+
+	addr := swarm.MustParseHexAddress("f30c0aa7e9e2a0ef4c9b1b750ebfeaeb7c7c24da700bb089da19a46e3677824b")
+
+	minStake := big.NewInt(1).String()
+	minStakedAmount := stakingcontract.MinimumStakeAmount
+	depositStake := func(address string, amount string) string {
+		return fmt.Sprintf("/stake/deposit/%s/%s", address, amount)
+	}
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		contract := stakingContractMock.New(
+			stakingContractMock.WithDepositStake(func(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error {
+				if stakedAmount.Cmp(minStakedAmount) == -1 {
```

Comment:
So adding logic of the actual contracts in the Mock function to test some logic is not required in API tests. These tests should only be concerned about the code in the `api` package. The stake amount check is not in API so we should not add tests for these things.

So basically you only need to test cases like:
- Send some invalid address and check parsing returns error
- Send some invalid amount and ensure parsing of amount fails
- Return different errors using the mock functions to test the if cases in your error handling scenarios